bolt/lib/Core/BinaryEmitter.cpp

```diff
@@ -319,6 +349,36 @@ bool BinaryEmitter::emitFunction(BinaryFunction &Function,
     Streamer.emitCodeAlignment(Function.getAlign(), &*BC.STI);
   }
 
+  if (size_t Padding = opts::padFunctionBefore(Function)) {
+    // Handle padFuncsBefore after the above alignment logic but before
+    // symbol addresses are decided; with the intent that the nops are
+    // not executed and the original alignment logic is preserved.
```

Comment:
I don't quite understand the comment about nops, since we are not using nops for padding.

Reply:
Thanks for spotting, residual text from earlier implementation.